src/main/java/uk/gov/pay/adminusers/service/InviteRouter.java

```diff
@@ -2,27 +2,35 @@
 
 import com.google.inject.Inject;
 import uk.gov.pay.adminusers.model.InviteType;
+import uk.gov.pay.adminusers.model.User;
 import uk.gov.pay.adminusers.persistence.entity.InviteEntity;
 
+import java.util.Optional;
+
 public class InviteRouter {
 
     private final InviteServiceFactory inviteServiceFactory;
+    private final UserServices userService;
 
     @Inject
-    public InviteRouter(InviteServiceFactory inviteServiceFactory) {
+    public InviteRouter(InviteServiceFactory inviteServiceFactory, UserServices userService) {
         this.inviteServiceFactory = inviteServiceFactory;
+        this.userService = userService;
     }
 
     public InviteCompleter routeComplete(InviteEntity inviteEntity) {
         switch (inviteEntity.getType()) {
             case SERVICE:
-            case NEW_USER_AND_NEW_SERVICE_SELF_SIGNUP:
                 return inviteServiceFactory.completeSelfSignupInvite();
-            case USER:
-            case EXISTING_USER_INVITED_TO_EXISTING_SERVICE:
-                return inviteServiceFactory.completeExistingUserInvite();
-            case NEW_USER_INVITED_TO_EXISTING_SERVICE:
-                return inviteServiceFactory.completeNewUserExistingServiceInvite();
+            case USER: {
+                Optional<User> user = userService.findUserByEmail(inviteEntity.getEmail());
+                if (user.isPresent()) {
+                    return inviteServiceFactory.completeExistingUserInvite();
+                }
+                else {
```

Comment:
If you do an intellij format code, it should put the `else` on the same line as the close brace